libs/ts-rest/fastify/src/lib/ts-rest-fastify.ts

```diff
@@ -0,0 +1,237 @@
+  registerRouter: <
+    T extends RecursiveRouterObj<TContract>,
+    TContract extends AppRouter
+  >(
+    contract: TContract,
+    routerImpl: T,
+    app: fastify.FastifyInstance,
+    options: RegisterRouterOptions = {
+      logInitialization: true,
+      jsonQuery: false,
+      responseValidation: false,
+    }
+  ) => {
+    recursivelyRegisterRouter(routerImpl, contract, [], app, options);
+  },
+});
+
+/**
+ *
+ * @param routeImpl - User's implementation of the route
+ * @param appRoute - the `ts-rest` contract for this route (e.g. with Path, params, query, body, etc.)
+ * @param fastify - the fastify instance to register the route on
+ */
+const registerRoute = <TAppRoute extends AppRoute>(
+  routeImpl: AppRouteImplementation<AppRoute>,
+  appRoute: TAppRoute,
+  fastify: fastify.FastifyInstance,
+  options: RegisterRouterOptions
+) => {
+  if (options.logInitialization) {
+    console.log(`[ts-rest] Initialized ${appRoute.method} ${appRoute.path}`);
+  }
+
+  fastify.route({
+    method: appRoute.method,
+    url: appRoute.path,
+    handler: async (request, reply) => {
+      const validationResults = validateRequest(
+        request,
+        reply,
+        appRoute,
+        options
+      );
+
+      if (validationResults === null) {
+        return;
+      }
+
+      const result = await routeImpl({
+        // eslint-disable-next-line @typescript-eslint/no-explicit-any
+        params: validationResults.paramsResult.data as any,
+        query: validationResults.queryResult.data,
+        // eslint-disable-next-line @typescript-eslint/no-explicit-any
+        headers: validationResults.headersResult.data as any,
+        request,
+        body: request.body,
+        reply,
+      });
+
+      const statusCode = result.status;
+
+      if (options.responseValidation) {
+        const response = validateResponse({
+          responseType: appRoute.responses[statusCode],
+          response: {
+            status: statusCode,
+            body: result.body,
+          },
+        });
+
+        return reply.status(statusCode).send(response.body);
+      }
+
+      return reply.status(statusCode).send(result.body);
+    },
+  });
+};
+
+/**
+ *
+ * @param routerImpl - the user's implementation of the router
+ * @param appRouter - the `ts-rest` contract for this router
+ * @param path - the path to the current router, e.g. ["posts", "getPosts"]
+ * @param fastify  - the fastify instance to register the route on
+ * @param options
+ */
+const recursivelyRegisterRouter = <T extends AppRouter>(
+  routerImpl: RecursiveRouterObj<T>,
+  appRouter: T,
+  path: string[],
+  fastify: fastify.FastifyInstance,
+  options: RegisterRouterOptions
+) => {
+  if (typeof routerImpl === 'object') {
+    for (const key in routerImpl) {
+      recursivelyRegisterRouter(
+        routerImpl[key] as unknown as RecursiveRouterObj<T>,
+        appRouter,
+        [...path, key],
```

Comment:
Would be easier, rather than having to propagate the path, if we instead pass the sub-routers in each iteration. So remove the `path` and just pass

```typescript
        routerImpl[key] as unknown as RecursiveRouterObj<T>,
        appRouter[key],
```

Will no longer need the `getValue` function as well